lib/ansible/modules/pip.py

```diff
@@ -611,6 +614,62 @@ def setup_virtualenv(module, env, chdir, out, err):
     return out, err, cmd
 
 
+def _normalize_vcs_packages(
+        module: AnsibleModule,
+        package_list: list[Package],
+        pip: list[str]
+) -> list[Package]:
+    """Return a list of packages with names extracted from metadata in VCS"""
+    vcs_packages = [pkg_str for pkg in package_list if _is_vcs_url(pkg_str := str(pkg))]
+
+    if not vcs_packages:
+        return package_list
+
+    # Unsetting 'FORCE_COLOR' with 'NO_COLOR' and TTY_COMPATIBLE settings, helps to
+    # prevent color codes from being injected into and corrupting the JSON output.
```

Comment:
This is probably more accurate. The behavior I was observing in my upstream bug report was ANSI codes that don't count as "color" because they controlled font boldness.
```suggestion
    # make pip output clean JSON without injected ANSI sequences that might corrupt it.
```